tests/installation/boot_encrypt.pm

```diff
@@ -12,14 +12,9 @@ use strict;
 use warnings;
 use base "installbasetest";
 use utils;
-use testapi qw(check_var get_var record_info);
-use version_utils qw(is_leap is_sle is_leap_micro is_sle_micro is_alp);
 
 sub run {
-    # With newer grub2 (in TW only currently), entering the passphrase in GRUB2
-    # is enough. The key is passed on during boot, so it's not asked for
-    # a second time.
-    return if is_boot_encrypted && !is_leap && !is_sle && !is_leap_micro && !is_sle_micro && !is_alp;
+    return if is_boot_encrypted && !need_passphrase_again;
```

Comment:
Shouldn't `is_boot_encrypted` be moved into `need_passphrase_again`?

Reply:
IMO, we can't do that, since in grub phase, if boot partition is not encrypted,we don't need to type passphrase. 

`need_passphrase_again` is used to check we need to type the passphrase after grub

Reply:
IIRC that's what I tried at first when I wrote that code but it's hardcoded in some places (maybe even yaml schedules) where this isn't easy to change.

Reply:
Agree with you, then it is safe to keep the check point